### PR TITLE
kodiPackages.infotagger: init at 0.0.7

### LINKED ISSUE #226613: kodi: youtube plugin is failing
### Describe the bug
When using Kodi with the YouTube plugin , I get the following error:

```
2023-04-17 11:25:54.137 T:6006    error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'ModuleNotFoundError'>
                                                   Error Contents: No module named 'infotagger'
                                                   Traceback (most recent call last):
                                                     File "/nix/store/wc8y1w830in0q9l1zyn5i161d3cmwk15-kodi-20.1-env/share/kodi/addons/plugin.video.youtube/resources/lib/startup.py", line 11, in <module>
                                                       from youtube_plugin.kodion import service
                                                     File "/nix/store/zycslxp5fkzgnfhly0yws6mdb9zzzvrw-kodi-youtube-7.0.1/share/kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/__init__.py", line 21, in <module>
                                                       from .impl import Context
                                                     File "/nix/store/zycslxp5fkzgnfhly0yws6mdb9zzzvrw-kodi-youtube-7.0.1/share/kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/__init__.py", line 12, in <module>
                                                       from .xbmc.xbmc_context import XbmcContext as Context
                                                     File "/nix/store/zycslxp5fkzgnfhly0yws6mdb9zzzvrw-kodi-youtube-7.0.1/share/kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_context.py", line 29, in <module>
                                                       from .xbmc_playlist import XbmcPlaylist
                                                     File "/nix/store/zycslxp5fkzgnfhly0yws6mdb9zzzvrw-kodi-youtube-7.0.1/share/kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_playlist.py", line 15, in <module>
                                                       from . import xbmc_items
                                                     File "/nix/store/zycslxp5fkzgnfhly0yws6mdb9zzzvrw-kodi-youtube-7.0.1/share/kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py", line 13, in <module>
                                                       from infotagger.listitem import ListItemInfoTag
                                                   ModuleNotFoundError: No module named 'infotagger'
                                                   -->End of Python script error report<--
```

### Steps To Reproduce
1. Use `pkgs.kodi-gbm.withPackages (p: with p; [ youtube ]);`
2. Start `kodi`

### Additional context
- The YouTube plugin used to work perfectly.
- https://github.com/anxdpanic/plugin.video.youtube/blame/master/addon.xml shows that the `inputstream.adaptive` dependency was added but that the `infotagger` dependency is an old one.
- The `infotagger` wrapper is here: https://github.com/jurialmunkey/script.module.infotagger

### Notify maintainers
@aanderse
@cpages
@edwtjo
@minijackson
@peterhoeg
@sephalon

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.23, NixOS, 23.05 (Stoat), 23.05.20230411.10193ca`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.14.1`
 - nixpkgs: `/nix/store/67iks6k4id77nzqqbc1w9bi4dcghxfp6-source`
```

### PR DESCRIPTION
**Pull request by dschrempf**
Fixes https://github.com/NixOS/nixpkgs/issues/226613.

Add `infotagger` plugin and use it.

Two notes on the `infotagger` plugin:

- I used the GitHub source, and not https://kodi.tv/addons/nexus/script.module.infotagger/ which links to http://ftp.halifax.rwth-aachen.de/xbmc/addons/nexus/script.module.infotagger/script.module.infotagger-0.0.7.zip. Not sure if this is a problem (or a feature).

- The `infotagger` plugin has a weird Python path `resources/modules`. I expect this path may change in the upstream module in the future (it is `lib` for all other modules as far as I can see).

I can watch YouTube videos here.

However, this PR does not address the relatively new dependency

```
<import addon="inputstream.adaptive" version="20.3.1"/>
```

I do not get any errors though. I am not sure how to proceed (open another issue?; ignore?).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dschrempf** [comment]
Two more cents:
- I think the automatic updater should scan for dependency changes, and fail if there are new or dependencies or if dependencies have been removed.
- This change was not caused by a dependency update of the YouTube addon, but by a change in base. (I think). The `infotagger` module seems to ensure some form of backwards compatibility.

File: pkgs/applications/video/kodi/addons/infotagger/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    updateScript = addonUpdateScript {
+      attrPath = "kodi.packages.infotagger";
+    };
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/jurialmunkey/script.module.infotagger";
+    description = "Wrapper for new Nexus InfoTagVideo ListItem methods to maintain backwards compatibility";
+    # No license.
+    # license = ?;
```
Comment (ErinvanderVeen) [inline pkgs/applications/video/kodi/addons/infotagger/default.nix:26]: License is GPL-3.0-or-later from https://kodi.tv/addons/nexus/script.module.infotagger/

**Review by ErinvanderVeen — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/kodi/addons/infotagger/default.nix (hunk 1 above)
Comment (dschrempf) [reply pkgs/applications/video/kodi/addons/infotagger/default.nix]: Thank you! This should be fixed!

**Review by ErinvanderVeen — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

**Comment by dschrempf** [comment]
Hi @SuperSandro2000. I was not completely sure if I should then split this into two commits (one adding the infotagger addon, the other one actually using it). As you see, I decided to go for one commit with a changed message, as you requested. Let me know if this is Ok.

**Comment by SuperSandro2000** [comment]
> one adding the infotagger addon, the other one actually using it

It is a new package so it should get an init commit.

**Comment by dschrempf** [comment]
I see. I didn't know this was so strict. Here you go!

**Pull request merged**